v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/transforms/EventMetadata.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (C) 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.elasticsearch.transforms;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.cloud.teleport.v2.elasticsearch.options.PubSubToElasticsearchOptions;
+import com.google.cloud.teleport.v2.elasticsearch.utils.Dataset;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * EventMetadata is used to insert metadata required by Elasticsearch.
+ * The metadata helps Elasticsearch to visualize events on the dashboards,
+ * also uniform message format is needed for data analytics.
+ */
+public class EventMetadata implements Serializable {
+
+    @JsonProperty("@timestamp")
+    private String timestamp;
+    @JsonProperty("agent")
+    private Agent agent;
+    @JsonProperty("data_stream")
+    private DataStream dataStream;
+    @JsonProperty("ecs")
+    private Ecs ecs;
+    @JsonProperty("message")
+    private String message;
+    @JsonProperty("service")
+    private Service service;
+    @JsonProperty("event")
+    private Event event;
+    @JsonIgnore
+    private String inputMessage;
+    @JsonIgnore
+    private JsonNode enrichedMessage;
+    @JsonIgnore
+    final ObjectMapper objectMapper = new ObjectMapper();
+
+    private EventMetadata(String inputMessage, PubSubToElasticsearchOptions pubSubToElasticsearchOptions) {
+        this.inputMessage = inputMessage;
+        event = new Event();
+        dataStream = new DataStream();
+        ecs = new Ecs();
+        service = new Service();
+        message = inputMessage;
+
+        try {
```

Comment:
Since inputMessage has potential to throw JsonProcessingException, perform the findTimestampValue check before declaring any instances such as  etc
event = new Event();
        dataStream = new DataStream();
        ecs = new Ecs();
        service = new Service();